src/macaron/build_spec_generator/cli_command_parser/maven_cli_parser.py

```diff
@@ -0,0 +1,594 @@
+        return "list[str]"
+
+
+# We intend to support Maven version 3.6.3 - 3.9
+MAVEN_OPTION_DEF: list[OptionDef] = [
+    MavenOptionalFlag(
+        short_name="-am",
+        long_name="--also-make",
+    ),
+    MavenOptionalFlag(
+        short_name="-amd",
+        long_name="--also-make-dependents",
+    ),
+    MavenOptionalFlag(
+        short_name="-B",
+        long_name="--batch-mode",
+    ),
+    MavenOptionalFlag(
+        short_name="-C",
+        long_name="--strict-checksums",
+    ),
+    MavenOptionalFlag(
+        short_name="-c",
+        long_name="--lax-checksums",
+    ),
+    MavenOptionalFlag(
+        short_name="-cpu",
+        long_name="--check-plugin-updates",
+    ),
+    MavenOptionalFlag(
+        short_name="-e",
+        long_name="--errors",
+    ),
+    MavenOptionalFlag(
+        short_name="-fae",
+        long_name="--fail-at-end",
+    ),
+    MavenOptionalFlag(
+        short_name="-ff",
+        long_name="--fail-fast",
+    ),
+    MavenOptionalFlag(
+        short_name="-fn",
+        long_name="--fail-never",
+    ),
+    MavenOptionalFlag(
+        short_name="-h",
+        long_name="--help",
+        dest="help_",
+    ),
+    MavenOptionalFlag(
+        short_name="-llr",
+        long_name="--legacy-local-repository",
+    ),
+    MavenOptionalFlag(
+        short_name="-N",
+        long_name="--non-recursive",
+    ),
+    MavenOptionalFlag(
+        short_name="-nsu",
+        long_name="--no-snapshot-updates",
+    ),
+    MavenOptionalFlag(
+        short_name="-ntp",
+        long_name="--no-transfer-progress",
+    ),
+    MavenOptionalFlag(
+        short_name="-npu",
+        long_name="--no-plugin-updates",
+    ),
+    MavenOptionalFlag(
+        short_name="-npr",
+        long_name="--no-plugin-registry",
+    ),
+    MavenOptionalFlag(
+        short_name="-o",
+        long_name="--offline",
+    ),
+    MavenOptionalFlag(
+        short_name="-q",
+        long_name="--quiet",
+    ),
+    MavenOptionalFlag(
+        short_name="-U",
+        long_name="--update-snapshots",
+    ),
+    MavenOptionalFlag(
+        short_name="-up",
+        long_name="--update-plugins",
+    ),
+    MavenOptionalFlag(
+        short_name="-v",
+        long_name="--version",
+    ),
+    MavenOptionalFlag(
+        short_name="-V",
+        long_name="--show-version",
+    ),
+    MavenOptionalFlag(
+        short_name="-X",
+        long_name="--debug",
+    ),
+    MavenGoalPhase(
+        long_name="goals",
+    ),
+    # TODO: we need to confirm whether one can provide
+    # -P or -pl multiple times and the values will be aggregate into a list of string
+    # The current implementation only consider one instance of -P or -pl.
+    # Where to begin:
+    # https://github.com/apache/maven/blob/maven-3.9.x/maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java
+    # https://github.com/apache/commons-cli/blob/master/src/main/java/org/apache/commons/cli/Parser.java
```

Comment:
It looks like this comment should be moved down a bit to where the related options are.